WordPress/Sniffs/Theme/NoDeregisterCoreScriptSniff.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * Sniff to check theme for deregister and register of core scripts (jquery)
+ *
+ * @category Theme
+ * @package  PHP_CodeSniffer
+ * @author   Simon Prosser <pross@pross.org.uk>
+ */
+class WordPress_Sniffs_Theme_NoDeregisterCoreScriptSniff implements PHP_CodeSniffer_Sniff {
+
+	/**
+	 * Core scripts to sniff for.
+	 *
+	 * @var array
+	 */
+	private $core_scripts = array(
+		'jquery',
+	);
+
+	/**
+	 * Returns an array of tokens this test wants to listen for.
+	 *
+	 * @return array
+	 */
+	public function register() {
+		return array(
+			T_STRING,
+			T_CONSTANT_ENCAPSED_STRING,
+			T_DOUBLE_QUOTED_STRING,
+			T_OPEN_PARENTHESIS,
+			T_CLOSE_PARENTHESIS,
+		);
```

Comment:
I'm wondering why you are sniffing for all these tokens. AFAICS see, all you really want to sniff for is `T_STRING`. Can you explain your reasoning behind this ?